src/helpers.js

```diff
@@ -135,7 +141,153 @@ const pathToChildRecursive = (basePath, child, courseData) => {
+  )
+  const files = courseData["course_files"].filter(
+    file => file["parent_uid"] === page["uid"]
+  )
+  const isParent = children.length > 0
+  const hasFiles = files.length > 0
+  return isParent || hasFiles ? "/_index.md" : ""
+}
+
+const resolveUids = (htmlStr, page, courseData) => {
+  try {
+    const pagePath = `${pathToChildRecursive(
+      path.join("courses", courseData["short_url"], "sections"),
+      page,
+      courseData
+    )}${getHugoPathSuffix(page, courseData)}`
+    Array.from(htmlStr.matchAll(/\.?\/?resolveuid\/.{0,32}/g)).forEach(
+      match => {
+        const url = match[0]
+        const urlParts = url.split("/")
+        const uid = urlParts[urlParts.length - 1]
+        const linkedPage = courseData["course_pages"].find(
+          coursePage => coursePage["uid"] === uid
+        )
+        const linkedFile = courseData["course_files"].find(
+          file => file["uid"] === uid
+        )
+        if (linkedPage) {
+          const linkPagePath = `${pathToChildRecursive(
+            path.join("courses", courseData["short_url"], "sections"),
+            linkedPage,
+            courseData
+          )}${getHugoPathSuffix(linkedPage, courseData)}`
+          htmlStr = htmlStr.replace(
+            url,
+            `${GETPAGESHORTCODESTART}${linkPagePath}${GETPAGESHORTCODEEND}`
+          )
+        }
+        if (linkedFile) {
+          if (linkedFile["file_type"] === "application/pdf") {
+            const pdfPath = `${pagePath.replace("/_index.md", "/")}${
+              linkedFile["id"]
+            }`
+            htmlStr = htmlStr.replace(
+              url,
+              `${GETPAGESHORTCODESTART}${pdfPath.replace(
+                ".pdf",
+                ""
+              )}${GETPAGESHORTCODEEND}`
+            )
+          } else {
+            htmlStr = htmlStr.replace(url, linkedFile["file_location"])
+          }
+        }
+      }
+    )
+  } catch (err) {
+    loggers.fileLogger.error(err.message)
+  }
+  return htmlStr
+}
+
+const resolveRelativeLinks = (htmlStr, courseData, log = false) => {
+  try {
+    Array.from(htmlStr.matchAll(/href="([^"]*)"/g)).forEach(match => {
+      const url = match[0].replace(`href="`, "").replace(`"`, "")
+      if (!url.includes("resolveuid") && url[0] === "/") {
+        const parts = url.split("/")
+        const courseId = parts[3]
+        if (courseId) {
+          const layers = parts.length - 4
+          const sections = []
+          let page = null
+          if (layers === 0) {
+            page = "index.htm"
+          } else if (layers === 1) {
+            page = parts[4]
+          } else {
+            for (let i = parts.length - layers; i < parts.length; i++) {
+              const section = parts[i]
+              if (i + 1 === parts.length) {
+                page = section
+              } else {
+                sections.push(section)
+              }
+            }
+          }
```

Comment:
I was thinking, with this logic to break down the URL and fiddle about with the course ID, it may be helpful to any future person reading this code to include a few comments explaining what a typical URL looks like, why we're saying that `layers = parts.length - 4`, etc, because without being familiar with the URLs in question it's pretty opaque what's going on here